client/src/pages/locations/GeoLocation.js

```diff
@@ -0,0 +1,122 @@
+import React from "react"
+import PropTypes from "prop-types"
+import { Button, Col, ControlLabel, FormGroup } from "react-bootstrap"
+import { Field } from "formik"
+import { Location } from "models"
+import REMOVE_ICON from "resources/delete.png"
+import * as FieldHelper from "components/FieldHelper"
+
+export const GEO_LOCATION_DISPLAY_TYPE = {
+  FORM_FIELD: "FORM_FIELD",
+  GENERIC: "GENERIC"
+}
+
+const GeoLocation = ({
+  lat,
+  lng,
+  setFieldTouched,
+  setFieldValue,
+  isSubmitting,
+  editable,
+  displayType
+}) => {
+  const setTouched = touched => {
+    setFieldTouched("lat", touched, false)
+    setFieldTouched("lng", touched, false)
+  }
+
+  if (!editable) {
+    const humanValue = (
+      <>
+        <span>{Location.parseCoordinate(lat) || "?"}</span>
+        <span>,&nbsp;</span>
+        <span>{Location.parseCoordinate(lng) || "?"}</span>
+      </>
+    )
+
+    if (displayType === GEO_LOCATION_DISPLAY_TYPE.GENERIC) return humanValue
+
+    if (displayType === GEO_LOCATION_DISPLAY_TYPE.FORM_FIELD) {
+      return (
+        <Field
+          name="location"
+          label="Latitude, Longitude"
+          component={FieldHelper.ReadonlyField}
+          humanValue={humanValue}
+        />
+      )
+    }
+
+    throw new Error(`Unknown displayType: '${displayType}' for GeoLocation`)
```

Comment:
This seems unnecessary, since the propTypes only allow for the two values above. Perhaps consider rewriting the two `if`s as a `switch` (and you could use the `throw …` as `default` case).

Reply:
Fixed by returning `humanValue` as default instead of throwing error. As you said it is unnecessary.